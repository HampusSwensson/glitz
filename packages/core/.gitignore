node_modules/
es/
server/