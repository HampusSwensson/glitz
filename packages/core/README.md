# ![Glitz](https://github.com/frenic/glitz/raw/master/glitz.svg?sanitize=true)

The core implementation for Glitz. Head over to [https://github.com/frenic/glitz](https://github.com/frenic/glitz) for details.
