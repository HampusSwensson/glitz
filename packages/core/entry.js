// Entry file for bundle

export { default as GlitzClient } from './src/client/GlitzClient.ts';
