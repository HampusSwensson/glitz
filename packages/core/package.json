{
  "name": "@glitz/core",
  "version": "1.1.5",
  "main": "./cjs/index.js",
  "module": "./jsm/index.js",
  "esnext": "./esnext/index.js",
  "types": "./esnext/index.d.ts",
  "license": "MIT",
  "author": "Fredrik Nicol <fredrik.nicol@gmail.com>",
  "homepage": "https://github.com/frenic/glitz",
  "repository": "git@github.com:frenic/glitz.git",
  "bugs": "https://github.com/frenic/glitz/issues",
  "side-effects": false,
  "dependencies": {
    "@glitz/type": "^1.1.4"
  },
  "scripts": {
    "clean": "rimraf cjs esnext jsm",
    "build": "tsc -p tsconfig.build.json && node ../../bundle.js src/index.ts --cjsx2 index --jsm index && node ../../bundle.js src/server.ts --cjs server --jsm server",
    "prepublish": "npm run clean & npm run build"
  },
  "files": [
    "cjs/",
    "esnext/",
    "jsm/",
    "server/"
  ]
}
