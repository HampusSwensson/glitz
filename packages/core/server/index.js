module.exports = require('../cjs/glitz.server');
