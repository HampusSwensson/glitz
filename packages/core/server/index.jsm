export * from '../jsm/server';
export { default } from '../jsm/server';
