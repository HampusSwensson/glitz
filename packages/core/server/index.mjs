export * from '../mjs/server';
export { default } from '../mjs/server';
