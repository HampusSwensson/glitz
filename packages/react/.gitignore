node_modules/
cjs/
es6/
jsm/