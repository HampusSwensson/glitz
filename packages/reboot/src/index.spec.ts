// tslint:disable max-classes-per-file

import { GlitzServer } from '@glitz/core';
import { GlitzProvider, styled, StyleProvider } from '@glitz/react';
import { mount } from 'enzyme';
import * as React from 'react';
import glitzReboot from '.';

describe('reboot', () => {
  it('renders input with reboot style', () => {
    const glitz = new GlitzServer();

    mount(
      React.createElement(
        GlitzProvider,
        { glitz },
        React.createElement(StyleProvider, { include: glitzReboot() }, React.createElement(styled.Input)),
      ),
    );

    expect(glitz.getStyleMarkup()).toMatchSnapshot();
  });
  it('overrides style', () => {
    const glitz = new GlitzServer();

    mount(
      React.createElement(
        GlitzProvider,
        { glitz },
        React.createElement(
          StyleProvider,
          { include: glitzReboot() },
          React.createElement(styled.P, {
            css: { marginBottom: '1em' },
          }),
        ),
      ),
    );

    expect(glitz.getStyleMarkup()).toMatchSnapshot();
  });
});
