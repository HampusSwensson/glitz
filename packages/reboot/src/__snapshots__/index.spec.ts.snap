// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reboot overrides style 1`] = `"<style data-glitz>.a{box-sizing:border-box}.d{margin-top:0}.e{margin-bottom:1em}.b:before{box-sizing:border-box}.c:after{box-sizing:border-box}</style>"`;

exports[`reboot renders input with reboot style 1`] = `"<style data-glitz>.a{box-sizing:border-box}.d{margin-left:0}.e{margin-right:0}.f{margin-top:0}.g{margin-bottom:0}.h{font-family:inherit}.i{font-size:inherit}.j{line-height:inherit}.k{overflow:visible}.b:before{box-sizing:border-box}.c:after{box-sizing:border-box}.l[type=\\"button\\"]{-webkit-appearance:button}.m[type=\\"reset\\"]{-webkit-appearance:button}.n[type=\\"submit\\"]{-webkit-appearance:button}.o[list]::-webkit-calendar-picker-indicator{display:none}.p[type=\\"date\\"]{-webkit-appearance:textarea}.q[type=\\"time\\"]{-webkit-appearance:textarea}.r[type=\\"datetime-local\\"]{-webkit-appearance:textarea}.s[type=\\"month\\"]{-webkit-appearance:textarea}.t::-webkit-datetime-edit{overflow:visible}.u::-webkit-datetime-edit{line-height:0}.v[type=\\"search\\"]{outline-offset:-2px}.w[type=\\"search\\"]{-webkit-appearance:textfield}.x::-webkit-search-decoration{font-style:inherit}.y::-webkit-search-decoration{font-variant:inherit}.z::-webkit-search-decoration{font-weight:inherit}.a0::-webkit-search-decoration{font-stretch:inherit}.a1::-webkit-search-decoration{font-size:inherit}.a2::-webkit-search-decoration{font-family:inherit}.a3::-webkit-search-decoration{line-height:inherit}.a4::-webkit-search-decoration{-webkit-appearance:button}.a5::-webkit-color-swatch-wrapper{padding-left:0}.a6::-webkit-color-swatch-wrapper{padding-right:0}.a7::-webkit-color-swatch-wrapper{padding-top:0}.a8::-webkit-color-swatch-wrapper{padding-bottom:0}</style>"`;
